### ISSUE TITLE
QUESTION: How to disable tab completion on foreign file systems?

### ISSUE THREAD

**Issue by ghost** [ohmyzsh/ohmyzsh#7348]
Tab completion was slow, and I traced it down to a foreign filesystem (Google Drive, mounted via fuse).  Thing is, I am not searching in those directories.  E.g.:

```
/home/me
|______/documents/
       |_________/blah/
|______/Google Drive/
```
typing:
```
$ ls ~/documents/blah/d<tab>
```
takes forever; I _suspected_ this, but I proved it by (1) observing the fuse `google-drive` process pegging the CPU, and (2) unmounting the fuse filesystems and observing that no lag occurred in completion.  

So, I found #6967, which also more or less confirms this except the answer was unsatisfactory: "we can't do anything about slow filesystems."  True enough, but what _would_ be helpful would be information about excluding specific directories from autocompletion.  I see that oh-my-zsh is already setting completion widgets, which I don't want to override with, e.g. [my own function](https://superuser.com/questions/585545/how-to-disable-zsh-tab-completion-for-nfs-dirs); is there documentation that explains how to do this (exclude specific directories)?

I believe this is caused by `zcompdump` caching the mounted drives. Removing it does fix the issue... until I reflexively <tab> in one of those directories again, and it gets cached again, and I have to remove the file again.  As I will not be able to stop myself reflexively use tab-completion in those directories, I'd prefer if oh-my-zsh just turned off tab completion for them.

Is this possible?

**Comment by bobdoah** [comment]
I'm having the same issue with mounted NFS exports.

**Comment by mcornella** [comment]
You can enable tracing of completion functions with `typeset -f -t _main_complete`, trigger the completion of the slow path and then watch where the execution ends up when the thing slows down. I believe this will be in the `_files` or `_path_files` functions, but just to be sure post a trace of the last lines to be executed so I can help you devise a hack.

**Comment by htfy96** [comment]
In my case, when I try to complete with `a/b/c/d/<tab>`, zsh got stuck at `compfiles -P tmp1 accex / 'm:{a-zA-Z}={A-Za-z} m:{a-zA-Z}={A-Za-z}' '' fake`, where `tmp1` is `a/b`. Both `a/b` and `a/b/c` are slow NFS directory with ~1K sub-directories.

**Comment by htfy96** [comment]
Maybe this isn't directed related, but https://unix.stackexchange.com/questions/162078/stop-zsh-from-completing-parent-directories works for me.

tl;dr: turn off [accept-exact-dirs](http://zsh.sourceforge.net/Doc/Release/Completion-System.html#index-accept_002dexact_002ddirs_002c-completion-style) with:

```
zstyle ':completion:*' accept-exact-dirs true
```

**Comment by mcornella** [comment]
Thanks for posting a possible solution, I'll close this.
